<a href="https://colab.research.google.com/github/NatalieAleksandrova2026/DTA_2026/blob/main/Homework/task_62" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 🐍 Python: Практикум

**Час виконання:** 3 години  
**Формат:** індивідуальна робота

---

## 📋 Що ви будете практикувати

| Розділ | Тема | Кількість завдань |
|---|---|---|
| 1 | `print`, `input`, змінні | 8 |
| 2 | `if / elif / else` (умови) | 10 |
| 3 | Цикл `for` з `range()` | 10 |
| 4 | Цикл `for` з рядками і списками | 10 |
| 5 | Цикл `while` | 8 |
| 6 | Списки (`list`) — методи та операції | 10 |
| 7 | 🔥 Комбіновані завдання | 6 |

**Усього:** 62 завдання

---

## 📌 Як працювати з зошитом

1. Читайте умову уважно — там може бути більше деталей, ніж здається
2. Не поспішайте — якщо завдання здається складним, почніть з простого варіанту, а потім ускладнюйте
3. Запускайте код часто (Shift + Enter), щоб перевіряти, що все працює
4. Якщо щось не виходить — перечитайте підказку (якщо є), погугліть, поекспериментуйте
5. **Помилки — це нормально!** Кожен програміст бачить їх сотні разів на день

> 💡 У завданнях позначено: ⭐ — легке, ⭐⭐ — середнє, ⭐⭐⭐ — складне

---

# Розділ 1. `print`, `input` та змінні

Базові завдання на введення-виведення та роботу зі змінними.

**Що треба знати:**
- `print(...)` — вивести на екран
- `input("текст: ")` — отримати ввід від користувача (повертає текст!)
- `int(...)`, `float(...)` — перетворити текст у число
- Арифметика: `+`, `-`, `*`, `/`, `**` (степінь), `%` (залишок)

### Завдання 1. Привітання ⭐

Запитайте у користувача його ім'я і виведіть привітання.

**Приклад:**
```
Як вас звати? Олена
Привіт, Олена!
```

In [ ]:
# Ваш код для завдання 1:
n = input("What is your name? ")
print(n)


What is your name? Olena
Olena


### Завдання 2. Сума двох чисел ⭐

Запитайте у користувача два числа і виведіть їхню суму.

**Приклад:**
```
Перше число: 7
Друге число: 12
Сума: 19
```

> 💡 **Підказка:** Не забудьте перетворити введення на число через `int(...)`.

In [ ]:
# Ваш код для завдання 2:
n1 = int(input("Перше число "))
n2 = int(input("Друге число "))
sum = n1 + n2
print("Сума ", sum)


Перше число 7
Друге число 12
Сума  19


### Завдання 3. Площа прямокутника ⭐

Запитайте довжину та ширину прямокутника і виведіть його площу.

In [ ]:
# Ваш код для завдання 3:
a = int(input("Enter length: "))
b = int(input("Enter width: "))
area = a * b
print(area)


Enter length: 6
Enter width: 8
48


### Завдання 4. Сантиметри в метри ⭐

Запитайте у користувача кількість сантиметрів і переведіть їх у метри.

**Приклад:**
```
Кількість см: 250
Це 2.5 м
```

In [ ]:
# Ваш код для завдання 4:
m = int(input("Кількість см: "))
n = m / 100
print("Це", n, "м")


Кількість см: 250
Це 2.5 м


### Завдання 5. Округлення оцінки ⭐⭐

Запитайте у користувача оцінку (число з крапкою, наприклад 4.7).
Виведіть її, округлену до цілого числа.

> 💡 **Підказка:** Використайте функцію `round()`. Не забудьте `float()` замість `int()`.

In [ ]:
# Ваш код для завдання 5:
n = float(input("Enter grade: "))
print(round(n))


Enter grade: 4.7
5


### Завдання 6. Ціна з ПДВ ⭐⭐

Запитайте у користувача ціну товару без ПДВ.
Розрахуйте та виведіть:
- суму ПДВ (20% від ціни)
- кінцеву ціну (ціна + ПДВ)

In [ ]:
# Ваш код для завдання 6:
p = float(input("Price: "))
s = p * 20 / 100 + p
print(s)


Price: 89
106.8


### Завдання 7. Зворотній відлік ⭐⭐

Запитайте у користувача число `n` і виведіть на одному рядку числа від `n` до 1, розділені пробілами.

**Приклад:**
```
Введіть число: 5
5 4 3 2 1
```

> 💡 **Підказка:** Можна використати `print(n, n-1, n-2, n-3, n-4)` — лише для цього завдання, без циклу.

In [ ]:
# Ваш код для завдання 7:
n = int(input("Enter number: "))
for i in range(n, 0, -1):
  print(i)



Enter number: 5
5
4
3
2
1


### Завдання 8. Обмін значень змінних ⭐⭐⭐

Створіть дві змінні `a = 10` і `b = 20`. Поміняйте їхні значення місцями (щоб `a` стало 20, а `b` — 10) і виведіть результат.

> 💡 **Підказка:** Можна через третю тимчасову змінну, або в Python є дуже короткий спосіб: `a, b = b, a`.

In [ ]:
# Ваш код для завдання 8:
a = 10
b = 20

In [ ]:
a, b = b, a

In [ ]:
c = a
a = b
b = c

10

In [ ]:
"""a = 10
b = 20
c = a
a = b
b = c
print(a, b)"""
a = 20
b = 30
a, b = b, a
print(a, b)


30 20


---

# Розділ 2. Умови `if / elif / else`

**Що треба знати:**
- `if умова:` — якщо умова правдива
- `elif умова:` — інакше якщо
- `else:` — у всіх інших випадках
- Оператори порівняння: `==`, `!=`, `>`, `<`, `>=`, `<=`
- Логічні оператори: `and`, `or`, `not`

### Завдання 9. Парне чи непарне ⭐

Запитайте у користувача число і виведіть, парне воно чи непарне.

> 💡 **Підказка:** Парне число ділиться на 2 без остачі: `n % 2 == 0`.

In [ ]:
# Ваш код для завдання 9:
n = int(input("Enter number: "))
if n % 2 == 0:
  print("Even number")
else:
  print("Odd number")

Enter number: 26
Even number


### Завдання 10. Додатнє, від'ємне чи нуль ⭐

Запитайте у користувача число і виведіть, додатнє воно, від'ємне чи дорівнює нулю.

In [ ]:
# Ваш код для завдання 10:
n = int(input("Enter number: "))
if n > 0:
  print ("Positive number")
elif n < 0:
  print("Negative number")
elif n == 0:
  print("zero")


Enter number: 0
zero


### Завдання 11. Більше з двох ⭐

Запитайте у користувача два числа і виведіть більше з них.
Якщо числа рівні — виведіть «числа рівні».

In [ ]:
# Ваш код для завдання 11:
n1 = int(input("Enter 1 number: "))
n2 = int(input("Enter 2 number: "))
if n1 > n2:
  print(n1)
elif n2 > n1:
  print(n2)
else:
  print("Equal numbers")


Enter 1 number: 7
Enter 2 number: 7
Equal numbers


### Завдання 12. Чи може людина голосувати ⭐

Запитайте у користувача вік. Виведіть:
- «Ви можете голосувати» — якщо вік ≥ 18
- «Ви ще малі для голосування» — якщо вік < 18
- «Некоректний вік» — якщо вік від'ємний

In [ ]:
# Ваш код для завдання 12:
a = int(input("Age: "))
if a >= 18 and a < 100:
  print("adult")
elif a < 18 and a > 0:
  print("child")
else:
  print("Invalid age")



Age: 8
child


### Завдання 13. Оцінка за бал ⭐⭐

Запитайте у користувача оцінку (від 0 до 100) і виведіть:
- 90+ → «Відмінно»
- 75-89 → «Добре»
- 60-74 → «Задовільно»
- 0-59 → «Не склав»
- інакше (від'ємні або >100) → «Некоректна оцінка»

In [ ]:
# Ваш код для завдання 13:
g = int(input("Enter grade: "))
if g >=90 and g <= 100:
  print("Excellent")
elif g >= 75 and g < 90:
  print("Good")
elif g >=60 and g < 75:
  print("C")
elif g >= 0 and g <60:
  print("Not accepted")
else:
  print("Invalid grade")

Enter grade: 120
Invalid grade


### Завдання 14. Найбільше з трьох ⭐⭐

Запитайте у користувача три числа і виведіть найбільше з них.

> 💡 **Підказка:** Можна або послідовно через `if/elif`, або скористатись вбудованою функцією `max()`.

In [ ]:
# Ваш код для завдання 14:
n1 = int(input("Enter number: "))
n2 = int(input("Enter number: "))
n3 = int(input("Enter number: "))
"""if n1 > n2 and n1 > n3:
  print(n1)
elif n2 > n1 and n2 > n3:
  print(n2)
elif n3 > n1 and n3 > n2:
  print(n3)
elif n1 == n2 and n1 > n3 or n1 == n3 and n1 > n2:
  print(n1)
elif n2 == n3 and n2 > n1:
  print(n2)
else:
  print("n1")"""
print(max(n1, n2, n3))

Enter number: 7
Enter number: 8
Enter number: 8
8


### Завдання 15. Високосний рік ⭐⭐

Запитайте у користувача рік і визначте, чи є він високосним.

**Правила:** Рік є високосним, якщо:
- ділиться на 4, АЛЕ
- НЕ ділиться на 100, АБО
- ділиться на 400

**Приклади:** 2024 — високосний, 2023 — ні, 1900 — ні, 2000 — високосний.

> 💡 **Підказка:** Використайте логічні оператори `and` та `or`. Умова: `(рік % 4 == 0 and рік % 100 != 0) or рік % 400 == 0`.

In [ ]:
# Ваш код для завдання 15:
y = int(input("Year: "))
if (y % 4 == 0 and y % 100 != 0) or y % 400 == 0:
  print("Leap year")
else:
  print("Common year")

Year: 2026
Common year


### Завдання 16. Пори року ⭐⭐

Запитайте у користувача номер місяця (1-12) і виведіть, до якої пори року він належить:
- 12, 1, 2 → зима
- 3, 4, 5 → весна
- 6, 7, 8 → літо
- 9, 10, 11 → осінь
- інакше → «Некоректний місяць»

In [ ]:
# Ваш код для завдання 16:
m = int(input("Month "))
if m == 12 or m == 1 or m == 2:
  print("Winter")
elif m == 3 or m == 4 or m == 5:
  print("Spring")
elif m == 6 or m == 7 or m == 8:
  print("Summer")
elif m == 9 or m == 10 or m == 11:
  print("Autumn")
else:
  print("Invalid month")

Month 13
Invalid month


### Завдання 17. Калькулятор ⭐⭐

Запитайте у користувача два числа і знак операції (`+`, `-`, `*`, `/`).
Виведіть результат відповідної операції.

**Приклад:**
```
Перше число: 10
Друге число: 3
Знак: *
Результат: 30
```

⚠️ Не забудьте перевірити ділення на 0!

In [ ]:
# Ваш код для завдання 17:
n1 = int(input("Number 1 "))
n2 = int(input("Number 2 "))
a = input("Operation sign ")
if a == "+":
  print(n1 + n2)
elif a == "-":
  print(n1 - n2)
elif a == "*":
  print(n1 * n2)
elif a == "/" and n2 != 0:
  print(n1 / n2)
elif a == "/"  and n2 == 0:
  print("Zero division mistake")
else:
  print("Invalid sign")


Number 1 8
Number 2 0
Operation sign /
Zero division mistake


### Завдання 18. Трикутник ⭐⭐⭐

Запитайте у користувача три довжини сторін і визначте:
1. Чи можна з них скласти трикутник? (правило: сума будь-яких двох сторін має бути більшою за третю)
2. Якщо можна — який це трикутник:
   - рівносторонній (всі сторони рівні)
   - рівнобедрений (дві сторони рівні)
   - різносторонній (всі різні)

> 💡 **Підказка:** Спочатку перевірте, чи трикутник взагалі існує. Потім вже визначайте тип через порівняння сторін.

In [ ]:
# Ваш код для завдання 18:
a = int(input("Сторона 1 "))
b = int(input("Сторона 2 "))
c = int(input("Сторона 3 "))
if a + b > c and b + c > a and a + c > b:
  print("трикутник існує")
  if a == b and b == c and a == c:
    print("рівносторонній")
  elif a == b or b == c or a == c:
    print("рівнобедрений")
  else:
    print("різносторонній")
else:
  print("трикутник не існує")


Сторона 1 3
Сторона 2 3
Сторона 3 4
трикутник існує
рівнобедрений


---

# Розділ 3. Цикл `for` з `range()`

**Що треба знати:**
- `range(n)` → 0, 1, ..., n-1
- `range(a, b)` → a, a+1, ..., b-1
- `range(a, b, step)` → a, a+step, ... (з кроком, не доходячи до b)

### Завдання 19. Числа від 1 до 20 ⭐

Виведіть всі числа від 1 до 20, кожне на новому рядку.

In [ ]:
# Ваш код для завдання 19:
for a in range(1, 21, 1):
  print(a)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20


### Завдання 20. Парні числа ⭐

Виведіть всі парні числа від 2 до 30.

> 💡 **Підказка:** Можна через `range(2, 31, 2)` або через перевірку `if n % 2 == 0`.

In [ ]:
# Ваш код для завдання 20:
"""for a in range(2, 31, 2):
  print(a)"""
for a in range(1, 31, 1):
  if a % 2 == 0:
    print(a)


2
4
6
8
10
12
14
16
18
20
22
24
26
28
30


### Завдання 21. Сума від 1 до 100 ⭐

Підрахуйте суму всіх чисел від 1 до 100 і виведіть результат. Очікуване число — 5050.

In [ ]:
# Ваш код для завдання 21:
print(sum(range(1, 101)))

5050


### Завдання 22. Таблиця множення ⭐

Запитайте у користувача число і виведіть таблицю множення цього числа на числа від 1 до 10.

**Приклад при введенні 7:**
```
7 * 1 = 7
7 * 2 = 14
...
7 * 10 = 70
```

In [ ]:
# Ваш код для завдання 22:
n = int(input("Number "))
for i in range(1, 11):
  print(n, "*", i, "=", n * i)



Number 5
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


### Завдання 23. Сума парних чисел ⭐⭐

Підрахуйте суму всіх парних чисел від 1 до 100.

In [ ]:
# Ваш код для завдання 23:
"""n = 0
for i in range(1, 101):
  if i % 2 == 0:
    n += i
print(n)"""
n = 0
for i in range(2, 101, 2):
  n += i
print(n)

2550


### Завдання 24. Факторіал ⭐⭐

Запитайте у користувача число `n` і обчисліть його факторіал (`n!`).

**Нагадування:** факторіал — це добуток всіх чисел від 1 до `n`.  
Наприклад, `5! = 1 * 2 * 3 * 4 * 5 = 120`.

> 💡 **Підказка:** Створіть змінну `result = 1`, а потім у циклі множте її на кожне число.

In [ ]:
# Ваш код для завдання 24:
n = int(input("Enter number: "))
result = 1
for i in range(1, n + 1, 1):
  result *= i
  print(result)



Enter number: 5
1
2
6
24
120


### Завдання 25. Степінь числа ⭐⭐

Запитайте у користувача два числа: основу `a` і степінь `n`.
Обчисліть `a` в степені `n`, НЕ використовуючи оператор `**`.

**Приклад:** `a = 3, n = 4` → `3 * 3 * 3 * 3 = 81`

> 💡 **Підказка:** Створіть змінну `result = 1` і помножте її на `a` рівно `n` разів.

In [ ]:
# Ваш код для завдання 25:
a = int(input("Enter number "))
n = int(input("Enter degree "))
result = 1
for i in range(1, n + 1):
  result *= a
print(result)

Enter number 8
Enter degree 9
134217728


### Завдання 26. Сума цифр числа ⭐⭐⭐

Запитайте у користувача число і знайдіть суму його цифр.

**Приклад:** число 1234 → сума 1+2+3+4 = 10

> 💡 **Підказка:** Найпростіший спосіб: перетворити число на рядок через `str()`, а потім перебрати символи у циклі і додавати `int(символ)` до суми.

In [ ]:
# Ваш код для завдання 26:
"""n = int(input("Enter number "))
s = 0
while n!= 0:
  s += n % 10
  n = n // 10
print(s)"""
n = str(input("Enter number "))
s = 0
k = len(n)
for i in range(0, k):
  s += int(n[i])
  print(s)

Enter number 1234
1
3
6
10


### Завдання 27. Прості числа ⭐⭐⭐

Виведіть всі прості числа від 2 до 50.

**Нагадування:** просте число — це число, яке ділиться тільки на 1 і саме на себе.

> 💡 **Підказка:** Для кожного числа `n` від 2 до 50 перевіряйте, чи має воно дільники від 2 до n-1. Можна використати змінну-прапорець `is_prime = True`.

In [ ]:
# Ваш код для завдання 27:
for i in range(2, 51):
  for j in range(2, i // 2 + 1):
    if i % j == 0:
      break
  else:
    print(i)


2
3
5
7
11
13
17
19
23
29
31
37
41
43
47


### Завдання 28. Малювання трикутника ⭐⭐⭐

Виведіть на екран прямокутний трикутник з зірочок. Висоту запитайте у користувача.

**Приклад при висоті 5:**
```
*
**
***
****
*****
```

> 💡 **Підказка:** Використайте вкладений цикл АБО операцію множення рядка: `'*' * n` повертає рядок з `n` зірочок.

In [ ]:
# Ваш код для завдання 28:
while True:
  h = int(input("Enter height "))
  if h < 2:
    print("Invalid number")
  else:
    for i in range(1, h + 1):
      print("*" * i)
    break



Enter height 7
*
**
***
****
*****
******
*******


---

# Розділ 4. Цикл `for` з рядками та списками

**Що треба знати:**
- `for letter in "Python":` — перебираємо літери в рядку
- `for item in [1, 2, 3]:` — перебираємо елементи списку
- `len(x)` — довжина рядка або списку
- Перевірка наявності: `if "a" in word:`

### Завдання 29. Довжина слова ⭐

Запитайте у користувача слово і виведіть кількість літер у ньому.

> 💡 **Підказка:** Використайте функцію `len()`.

In [ ]:
# Ваш код для завдання 29:
n = input("Word: ")
k = len(n)
print(k)

Word: Beautiful
9


### Завдання 30. Виведення кожної літери ⭐

Запитайте у користувача слово і виведіть кожну літеру на новому рядку.

In [ ]:
# Ваш код для завдання 30:
n = input("Word: ")
for letter in n:
  print(letter)

Word: happy
h
a
p
p
y


### Завдання 31. Підрахунок голосних ⭐⭐

Запитайте у користувача слово і підрахуйте кількість голосних літер у ньому.

Голосні: `а е и і о у я ю є ї`.

> 💡 **Підказка:** Створіть лічильник `count = 0`. У циклі перевіряйте: `if letter in "аеиіоуяюєї":` і збільшуйте лічильник.

In [ ]:
# Ваш код для завдання 31:
n = input("Word: ")
count = 0
for letter in n.lower():
  if letter in "аеиіоуяюєї":
    count += 1
print(count)

Word: Україна
4


### Завдання 32. Перевернуте слово ⭐⭐

Запитайте у користувача слово і виведіть його у зворотному порядку.

**Приклад:** `Python` → `nohtyP`

> 💡 **Підказка:** Створіть порожній рядок `result = ''` і у циклі додавайте кожну літеру НА ПОЧАТОК: `result = letter + result`.

In [ ]:
# Ваш код для завдання 32:
n = input("Word: ")
result = ''
for letter in n:
  result = letter + result
print(f"Перевернуте слово: {result}")

Word: Python
Перевернуте слово: nohtyP


### Завдання 33. Підрахунок певної літери ⭐⭐

Запитайте у користувача слово і літеру. Виведіть, скільки разів ця літера зустрічається у слові.

In [ ]:
# Ваш код для завдання 33:
w = input("Word ")
l = input("Letter ")
count = 0
for char in w.lower():
  if char == l.lower():
    count += 1
print(f"Letter {l} is in word {count} times")



Word happy
Letter p
Letter p is in word 2 times


### Завдання 34. Сума елементів списку ⭐

Дано список `numbers = [12, 7, 18, 3, 25, 9, 14]`. Знайдіть і виведіть суму всіх його елементів. Очікуваний результат: 88.

In [ ]:
# Ваш код для завдання 34:
numbers = [12, 7, 18, 3, 25, 9, 14]
count = 0
for n in numbers:
  count += n
print(count)


88


In [ ]:
# Дано:
numbers = [12, 7, 18, 3, 25, 9, 14]
# Ваш код:


### Завдання 35. Найбільший і найменший ⭐⭐

Дано список `numbers = [42, 17, 89, 3, 56, 91, 24, 8, 73]`. Знайдіть найбільший і найменший елементи **без використання `max()` та `min()`**.

> 💡 **Підказка:** Створіть змінні `maximum = numbers[0]` і `minimum = numbers[0]`, а потім у циклі порівнюйте.

In [ ]:
# Ваш код для завдання 35:


In [ ]:
numbers = [42, 17, 89, 3, 56, 91, 24, 8, 73]
maximum = numbers[0]
minimum = numbers[0]
for i in range(len(numbers)):
  if numbers[i] > maximum:
    maximum = numbers[i]
  if numbers[i] < minimum:
    minimum = numbers[i]
print(str(maximum)+", "+str(minimum))

91, 3


### Завдання 36. Тільки додатні ⭐⭐

Дано список з додатними та від'ємними числами. Виведіть лише додатні.

In [ ]:
# Ваш код для завдання 36:


In [ ]:
numbers = [12, -5, 7, -18, 25, 0, -3, 9, -14, 30]
for n in numbers:
  if n > 0:
    print(n, end= " ")


12 7 25 9 30 

### Завдання 37. Середній бал ⭐⭐

Дано список оцінок студентів. Підрахуйте середній бал.

> 💡 **Підказка:** Середнє = сума / кількість. Кількість можна отримати через `len()`.

In [ ]:
# Ваш код для завдання 37:


In [ ]:
grades = [85, 72, 90, 65, 78, 88, 92, 70, 81, 76]
total = 0
for grade in grades:
  total += grade
  count = len(grades)
  avg_grade = total / count
print(f"Average Grade: {avg_grade}")




### Завдання 38. Найдовше слово ⭐⭐⭐

Дано список слів. Знайдіть і виведіть найдовше слово.

> 💡 **Підказка:** Створіть змінну `longest = words[0]`, а потім у циклі порівнюйте через `len()`.

In [ ]:
# Ваш код для завдання 38:


In [ ]:
words = ["кіт", "програмування", "Python", "школа", "інформатика", "олівець"]
longest = words[0]
for i in range(len(words)):
  if len(words[i]) > len(longest):
    longest = words[i]
print(longest)


програмування


---

# Розділ 5. Цикл `while`

**Що треба знати:**
- `while умова:` — повторюємо, поки умова правдива
- ⚠️ Завжди має бути щось, що змінює умову — інакше нескінченний цикл!
- Використовуємо `while`, коли наперед не відомо, скільки разів повторювати

### Завдання 39. Лічильник до 10 ⭐

За допомогою циклу `while` виведіть числа від 1 до 10.

In [ ]:
# Ваш код для завдання 39:
num = 0
while num <= 10:
  num += 1
  print(num, end= " ")


1 2 3 4 5 6 7 8 9 10 11 

### Завдання 40. Зворотній відлік ⭐

За допомогою циклу `while` виведіть числа від 10 до 1, потім надрукуйте «Старт!».

In [ ]:
# Ваш код для завдання 40:
num = 11
while num > 1:
  num -= 1
  print(num, end= " ")
print("Start!")


10 9 8 7 6 5 4 3 2 1 Start!


### Завдання 41. Подвоєння ⭐⭐

Починаючи з числа 1, подвоюйте його, поки воно менше 1000. Виводьте кожне значення.

**Очікуваний результат:** 1, 2, 4, 8, 16, 32, 64, 128, 256, 512.

In [ ]:
# Ваш код для завдання 41:
num = 1
while num < 1000:

  print(num, end= " ")
  num *= 2



1 2 4 8 16 32 64 128 256 512 

### Завдання 42. Сума введених чисел ⭐⭐

Просіть користувача вводити числа, доки він не введе 0.
Після цього виведіть суму всіх введених чисел (нуль не враховується).

> 💡 **Підказка:** Створіть змінну `total = 0` і змінну `number = -1` (щоб умова `while number != 0:` спрацювала на початку). Усередині циклу запитуйте число і додавайте до суми.

In [ ]:
# Ваш код для завдання 42:
number = -1
total = 0
while number != 0:
  number = int(input("Enter number: "))
  total += number
print(total)

Enter number: 8
Enter number: 7
Enter number: 99
Enter number: 3
Enter number: 0
117


### Завдання 43. Кількість цифр у числі ⭐⭐

Запитайте у користувача додатнє ціле число і підрахуйте, скільки в ньому цифр.

**Приклад:** число 12345 → 5 цифр.

> 💡 **Підказка:** Поки число більше за 0, ділимо його націло на 10 (`n = n // 10`) і збільшуємо лічильник на 1.

In [ ]:
# Ваш код для завдання 43:
n = int(input("Enter number: "))
count = 0
while n > 0:
  n = n // 10
  count += 1
print(count)

Enter number: 1488
4


### Завдання 44. Перевірка пароля ⭐⭐

Створіть змінну `correct_password = "qwerty123"`.
У циклі `while` запитуйте у користувача пароль, доки він не введе правильний.
Після успішного введення виведіть «Доступ дозволено».

In [ ]:
# Ваш код для завдання 44:
correct_password = "qwerty123"
while True:
  password = input("Enter password: ")
  if password == correct_password:
    print("Correct password")
    break


Enter password: gghfjj67788
Enter password: dgghj
Enter password: qwerty123
Correct password


### Завдання 45. Сума цифр через while ⭐⭐⭐

Запитайте у користувача додатнє ціле число і знайдіть суму його цифр, використовуючи `while`.

**Приклад:** число 1234 → сума 1+2+3+4 = 10.

> 💡 **Підказка:** У циклі: беремо останню цифру через `n % 10`, додаємо до суми, потім відкидаємо її через `n = n // 10`. Цикл працює, поки `n > 0`.

In [ ]:
# Ваш код для завдання 45:
n = int(input("Enter number: "))
total = 0
while n > 0:
  total += n % 10
  n = n // 10

print(total)

Enter number: 18777
30


### Завдання 46. Найбільший спільний дільник ⭐⭐⭐

Запитайте у користувача два додатні числа і знайдіть їхній найбільший спільний дільник (НСД) за алгоритмом Евкліда.

**Алгоритм:** доки `b ≠ 0`: одночасно `a, b = b, a % b`. Коли `b = 0`, відповідь — це `a`.

**Приклад:** НСД(48, 36) = 12.

In [ ]:
# Ваш код для завдання 46:
n1 = int(input("Enter number 1: "))
n2 = int(input("Enter number 2: "))
while n1 % n2 != 0:
  n1, n2 = n2, n1 % n2
print(n2)



Enter number 1: 48
Enter number 2: 36
12


---

# Розділ 6. Списки (`list`) — методи та операції

**Що треба знати:**
- Створення: `[1, 2, 3]`, `[]` (порожній)
- Доступ: `lst[0]`, `lst[-1]` (останній)
- Довжина: `len(lst)`
- Методи: `.append(x)`, `.insert(i, x)`, `.remove(x)`, `.pop()`, `.sort()`, `.reverse()`
- Перевірка: `x in lst`

### Завдання 47. Створення списку ⭐

Створіть список з 5 ваших улюблених фільмів і виведіть його.

In [ ]:
# Ваш код для завдання 47:
movies = ["TVD", "Twilight", "Ladybug", "Titanic", "Up"]
print(movies)

['TVD', 'Twilight', 'Ladybug', 'Titanic', 'Up']


### Завдання 48. Додавання у список ⭐

Створіть порожній список. У циклі `for` додайте до нього числа від 1 до 10. Виведіть результат.

> 💡 **Підказка:** Використайте метод `.append()`.

In [ ]:
# Ваш код для завдання 48:
lst = []
for i in range(1, 11):
  lst.append(i)
print(lst)


[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


### Завдання 49. Перший і останній ⭐

Дано список. Виведіть його перший і останній елементи.

In [ ]:
# Ваш код для завдання 49:


In [ ]:
colors = ["червоний", "жовтий", "зелений", "синій", "фіолетовий"]
print(str(colors[0])+",", colors[len(colors)-1])


червоний, фіолетовий


### Завдання 50. Список покупок ⭐⭐

Створіть список покупок з 3-4 товарів. Виконайте такі дії та виводьте список після кожної:
1. Додайте новий товар у кінець
2. Вставте новий товар на початок
3. Видаліть один товар за назвою
4. Замініть товар на іншому місці на щось нове

In [ ]:
# Ваш код для завдання 50:
lst = ["apple", "kiwi", "milk", "olives"]
lst.append("ham")
print(lst)
lst.insert(0, "fish")
print(lst)
if "ham" in lst:
  lst.remove("ham")
print(lst)
for x in range(0, len(lst)):
  if lst[x] == "milk":
    lst[x] = "lemon"
print(lst)




['apple', 'kiwi', 'milk', 'olives', 'ham']
['fish', 'apple', 'kiwi', 'milk', 'olives', 'ham']
['fish', 'apple', 'kiwi', 'milk', 'olives']
['fish', 'apple', 'kiwi', 'lemon', 'olives']


### Завдання 51. Сортування і реверс ⭐⭐

Дано список чисел. Відсортуйте його за зростанням, виведіть. Потім переверніть, виведіть.

In [ ]:
# Ваш код для завдання 51:


In [ ]:
numbers = [42, 17, 89, 3, 56, 91, 24, 8, 73]
numbers.sort()
print(numbers)
numbers.reverse()
print(numbers)


[3, 8, 17, 24, 42, 56, 73, 89, 91]
[91, 89, 73, 56, 42, 24, 17, 8, 3]


### Завдання 52. Парні і непарні в окремі списки ⭐⭐

Дано список чисел. Створіть два нові списки: один з парними числами, другий з непарними. Виведіть обидва.

In [ ]:
# Ваш код для завдання 52:


In [ ]:
numbers = [12, 7, 18, 3, 25, 9, 14, 21, 8, 11, 6, 33]
lst1 = []
lst2 = []
for n in range(0, len(numbers)):
  if numbers[n] % 2 == 0:
    lst1.append(numbers[n])
  else:
    lst2.append(numbers[n])
print(lst1, lst2)



[12, 18, 14, 8, 6] [7, 3, 25, 9, 21, 11, 33]


### Завдання 53. Унікальні елементи ⭐⭐

Дано список з повторами. Створіть новий список, у якому кожен елемент зустрічається лише один раз.

> 💡 **Підказка:** Створіть порожній список `unique = []`. Перебирайте оригінальний список і додавайте елемент тільки якщо його ЩЕ НЕМАЄ в `unique`.

In [ ]:
# Ваш код для завдання 53:


In [ ]:
items = ["яблуко", "банан", "яблуко", "груша", "банан", "слива", "яблуко", "груша"]
unique = []
for el in items:
  if el not in unique:
    unique.append(el)
print(unique)


['яблуко', 'банан', 'груша', 'слива']


### Завдання 54. Підрахунок входжень ⭐⭐

Дано список і елемент. Підрахуйте, скільки разів цей елемент зустрічається у списку **без використання методу `.count()`**.

In [ ]:
# Ваш код для завдання 54:


In [ ]:
words = ["кіт", "пес", "кіт", "хом'як", "кіт", "пес", "папуга", "кіт"]
target = "кіт"
count = 0
for t in words:
  if t == target:
    count += 1
print(count)
# count = words.count(target)
# print(count)


4


### Завдання 55. Об'єднання списків ⭐⭐⭐

Дано два списки. Створіть новий список, який містить:
1. Спочатку елементи першого
2. Потім — елементи другого
3. Після — лише ті елементи з обох списків, які зустрічаються в обох

**Приклад:** `[1, 2, 3]` + `[2, 3, 4]` → `[1, 2, 3, 2, 3, 4, 2, 3]` (бо 2 і 3 — спільні).

In [ ]:
# Ваш код для завдання 55:


In [ ]:
list_a = [1, 2, 3, 4, 5]
list_b = [3, 4, 5, 6, 7]
unique = []
lst1 = []
for el in list_a:
  lst1.append(el)
for el in list_b:
  lst1.append(el)
print(lst1)
for el in list_a:
  if el in list_b:
    if el not in unique:
      unique.append(el)
print(unique)


[1, 2, 3, 4, 5, 3, 3, 5, 3, 4, 5, 6, 7]
[3, 4, 5]


### Завдання 56. Сортування без `.sort()` ⭐⭐⭐

Дано список чисел. Відсортуйте його за зростанням **без використання методу `.sort()` та функції `sorted()`**.

> 💡 **Підказка:** Найпростіший алгоритм — «бульбашкове сортування»: у вкладеному циклі порівнюємо сусідні елементи і міняємо їх місцями, якщо ліворуч стоїть більший. Повторюємо, поки не пройдемо весь список без жодної заміни.

In [ ]:
# Ваш код для завдання 56:


In [ ]:
numbers = [42, 17, 89, 3, 56, 91, 24]
for i in range(0, len(numbers)):
  for n in range(0, len(numbers) - i - 1):
    if numbers[n] > numbers[n+1]:
      numbers[n], numbers[n+1] = numbers[n+1], numbers[n]
print(numbers)
# numbers.sort()
# print(numbers)





[3, 17, 24, 42, 56, 89, 91]


---

# Розділ 7. 🔥 Комбіновані завдання

Тут вам знадобиться поєднати кілька різних тем у одному завданні. Це завдання, найбільш близькі до реальних задач, які доводиться розв'язувати програмістам.

### Завдання 57. Меню кафе ⭐⭐

Дано список страв і список їхніх цін (відповідні позиції).
1. Виведіть меню у форматі «Назва — Ціна грн».
2. Підрахуйте і виведіть середню ціну.
3. Виведіть страви, ціна яких вища за середню.

> 💡 **Підказка:** Щоб одночасно перебирати два списки, використовуйте індекс: `for i in range(len(dishes)):`, далі `dishes[i]` і `prices[i]`.

In [ ]:
# Ваш код для завдання 57:


In [ ]:
dishes = ["Борщ", "Вареники", "Салат", "Стейк", "Десерт", "Чай"]
prices = [85, 95, 70, 320, 110, 35]
count = 0
for i in range(0, len(dishes)):
  count += prices[i]
  print(dishes[i] + " - " + str(prices[i]) + " грн")
average = count / len(prices)
print("Average price:", average)
for a in range (0, len(dishes)):
  if prices[a] > average:
    print(dishes[a])



Борщ - 85 грн
Вареники - 95 грн
Салат - 70 грн
Стейк - 320 грн
Десерт - 120 грн
Чай - 35 грн
Average price: 120.83333333333333
Стейк


### Завдання 58. Шифр Цезаря ⭐⭐⭐

Запитайте у користувача слово (українською або англійською) і число `shift`.
Зашифруйте слово: кожну літеру замініть на ту, що стоїть на `shift` позицій далі в алфавіті.

**Приклад:** слово `abc`, `shift = 2` → `cde`.

> 💡 **Підказка:** Використайте функції `ord(letter)` (повертає код символу) і `chr(num)` (повертає символ за кодом). Для кожної літери: новий код = ord(letter) + shift. Для простоти можна не обробляти переходи через кінець алфавіту.

In [ ]:
# Ваш код для завдання 58:
t = input("Enter text: ")
n = int(input("Enter code: "))
l = ""
for i in range(0, len(t)):
  num = ord(t[i])
  l += chr(num+n)

print(l)

Enter code: 2
jcrr{


### Завдання 59. Гра «Вгадай число» ⭐⭐⭐

Створіть змінну `secret = 42` (загадане число). У циклі `while` запитуйте у користувача число:
- Якщо введене число менше за загадане — пишіть «Більше»
- Якщо більше — «Менше»
- Якщо вгадав — пишіть «Вітаю!» і завершуйте цикл
- Підрахуйте кількість спроб і виведіть її в кінці

> 💡 **Підказка:** Використайте змінну-прапорець `guessed = False` як умову циклу. Коли користувач вгадав — `guessed = True`.

In [ ]:
# Ваш код для завдання 59:
secret = 42
count = 0
while True:
  n = int(input("Enter number: "))
  count += 1
  if n < secret:
    print("Більше")
  elif n > secret:
    print("Менше")
  else:
    print("Вітаю!")
    break
print(count)

Enter number: 23
Більше
Enter number: 35
Більше
Enter number: 44
Менше
Enter number: 89
Менше
Enter number: 42
Вітаю!
5


### Завдання 60. Аналіз тексту ⭐⭐⭐

Дано рядок з реченням. Виконайте такий аналіз і виведіть результати:
1. Кількість літер (без пробілів)
2. Кількість слів
3. Кількість голосних літер
4. Найдовше слово

> 💡 **Підказка:** Метод `.split()` розбиває рядок на список слів. Наприклад, `'hello world'.split()` повертає `['hello', 'world']`.

In [ ]:
# Ваш код для завдання 60:


In [ ]:
text = "програмування на python це цікаво і корисно"
# Ваш код:
words = text.split(" ")
l = 0
for char in text:
  if char != " ":
    l += 1
word_count = len(words)
vowels = 0
for char in text.lower():
  if char in "аеиіоуяюєїaeiouy":
    vowels += 1
longest = words[0]
for word in words:
  if len(word) > len(longest):
    longest = word
print(f"Літер без пробілів: {l}")
print(f"Слів: {word_count}")
print(f"Голосних: {vowels}")
print(f"Найдовше слово: {longest}")


Літер без пробілів: 37
Слів: 7
Голосних: 16
Найдовше слово: програмування


### Завдання 61. Магазин: статистика продажів ⭐⭐⭐

Дано список продажів за тиждень (по одному числу на день, починаючи з понеділка).
Виведіть:
1. Загальну виручку за тиждень
2. Середню виручку на день
3. Найкращий день (з найбільшою виручкою) — назвою (Понеділок, Вівторок...)
4. Кількість днів, коли виручка була вищою за середню

> 💡 **Підказка:** Створіть список з назвами днів: `days = ["Понеділок", "Вівторок", ...]`. Індекс найкращого дня знайдіть у циклі (або через `sales.index(max(sales))`).

In [ ]:
# Ваш код для завдання 61:


In [ ]:
sales = [3200, 2800, 4100, 3500, 5200, 7800, 6500]
# Ваш код:
days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
total = 0
for num in sales:
  total += num
avg_total = total / len(sales)
best_day = 0
for i in range(len(sales)):
  if sales[i] > sales[best_day]:
    best_day = i
count = 0
for num in sales:
  if num > avg_total:
    count += 1
print(f"Загальна виручка: {total}")
print(f"Середня виручка: {avg_total:.2f}")
print(f"Найкращий день: {days[best_day]}")
print(f"Днів вище середнього: {count}")





Загальна виручка: 33100
Середня виручка: 4728.57
Найкращий день: Saturday
Днів вище середнього: 3


### Завдання 62. Турнірна таблиця ⭐⭐⭐

Дано два списки: команди та їхні очки (відповідні позиції).
Виведіть турнірну таблицю, ВІДСОРТОВАНУ за спаданням очок, у такому форматі:
```
1. Шахтар — 67 очок
2. Динамо — 64 очки
...
```

> 💡 **Підказка:** Найпростіший спосіб: у циклі знаходьте команду з максимальними очками, виводьте її, потім видаляйте цю пару (команду і очки) з обох списків. Повторюйте, поки списки не спорожніють.

In [ ]:
# Ваш код для завдання 62:


In [ ]:
teams = ["Динамо", "Шахтар", "Зоря", "Дніпро", "Ворскла", "Олександрія"]
points = [64, 67, 52, 48, 39, 45]
# Ваш код:
place = 1
while len(points) > 0:
  i = points.index(max(points))
  print(f"{place}. {teams[i]} — {points[i]} очок")
  teams.pop(i)
  points.pop(i)
  place += 1

1. Шахтар — 67 очок
2. Динамо — 64 очок
3. Зоря — 52 очок
4. Дніпро — 48 очок
5. Олександрія — 45 очок
6. Ворскла — 39 очок
